test/source/tests/setup.ts

```diff
@@ -554,6 +556,167 @@ AN8G3r5Htj8olot+jm9mIa5XLXWzMNUZgg==
+      await gmailPage.close();
+      MOCK_KM_UPDATING_KEY.privateKeys = [{ decryptedPrivateKey: testConstants.unprotectedPrvKey }];
+      gmailPage = await browser.newPage(t, dummyGmailUrl, undefined, extraAuthHeaders);
+      await gmailPage.waitAll('@dialog-passphrase');
+      {
+        const passphraseDialog = await gmailPage.getFrame(['passphrase.htm']);
+        await passphraseDialog.waitAndType('@input-pass-phrase', 'g00D_pa$$worD-But_Different');
+        await passphraseDialog.waitAndClick('@action-confirm-pass-phrase-entry');
+        // todo: how to wait properly
+        await passphraseDialog.waitForContent('@input-pass-phrase', /^$/);
+        expect(await passphraseDialog.attr('@input-pass-phrase', 'placeholder')).to.eq('Please try again');
+      }
+      await ComposePageRecipe.cancelPassphraseDialog(gmailPage, 'keyboard');
+      await PageRecipe.noToastAppears(gmailPage);
+      const set11 = await retrieveAndCheckKeys(2);
+      expect(set11.map(entry => entry.prv.id)).to.eql(['392FB1E9FF4184659AB6A246835C0141B9ECF536', 'FAFB7D675AC74E87F84D169F00B0115807969D75']);
+      await gmailPage.close();
+      // 11. EKM returns a new third key, we enter a passphrase matching an existing key, update happens
+      gmailPage = await browser.newPage(t, dummyGmailUrl, undefined, extraAuthHeaders);
+      await gmailPage.waitAll('@dialog-passphrase');
+      {
+        const passphraseDialog = await gmailPage.getFrame(['passphrase.htm']);
+        await passphraseDialog.waitForContent('@passphrase-text', 'Enter FlowCrypt pass phrase to keep your account keys up to date');
+        await passphraseDialog.waitAndType('@input-pass-phrase', passphrase);
+        await passphraseDialog.waitAndClick('@action-confirm-pass-phrase-entry');
+      }
+      await gmailPage.waitTillGone('@dialog-passphrase');
+      await PageRecipe.waitForToastToAppearAndDisappear(gmailPage, 'Account keys updated');
+      const set12 = await retrieveAndCheckKeys(3);
+      expect(set12.map(entry => entry.prv.id)).to.eql([
+        '392FB1E9FF4184659AB6A246835C0141B9ECF536',
+        'FAFB7D675AC74E87F84D169F00B0115807969D75',
+        '277D1ADA213881F4ABE0415395E783DC0289E2E2'
+      ]);
+      await InboxPageRecipe.finishSessionOnInboxPage(gmailPage);
+      await gmailPage.close();
+      // 12. Forget the passphrase, EKM sends a broken key
+      // todo: we should probably update the valid keys?
+      MOCK_KM_UPDATING_KEY.privateKeys = [
+        { decryptedPrivateKey: await updateAndArmorKey(mainKey10[0]) }, // update the main key
+        // only include a half of another armored key
+        { decryptedPrivateKey: testConstants.unprotectedPrvKey.substring(0, testConstants.unprotectedPrvKey.length / 2) }
+      ];
+      gmailPage = await browser.newPage(t, dummyGmailUrl, undefined, extraAuthHeaders);
+      await PageRecipe.waitForToastToAppearAndDisappear(gmailPage, 'Could not update keys from EKM due to error:');
```

Comment:
@tomholub This test covers almost all of the cases from https://github.com/FlowCrypt/flowcrypt-ios/issues/841 (except 500 error yet)
I have some questions: 
1) do we want to display EKM errors as a toast, notification or modal?
Toast looks like this (by default, at the bottom of the page)
![image](https://user-images.githubusercontent.com/23335944/178025133-3e1eaa88-dc77-4278-9431-b4628f19dc5d.png)
A notification looks like this (at the top of the gmail page):
![image](https://user-images.githubusercontent.com/23335944/178025580-398d23e3-caee-4fc3-9603-f001c23e2a71.png)
2) What should happen if EKM sends 2 keys: one valid updated key and one broken key.
We should probably update the key(s) that can be updated and also show the error.
For this case, should we use toast for success and notification for errors? (Afaik only one toast at a time is supported).
3) There is a possibility of collision between more than one passphrase frame requests -- one can be coming from a `compose` frame, and another from the key update.  Should we consider emulating such a rare condition in a test and figure out a resolution?

Reply:
> What should happen if EKM sends 2 keys: one valid updated key and one broken key.
We should probably update the key(s) that can be updated and also show the error.
> For this case, should we use toast for success and notification for errors? (Afaik only one toast at a time is supported).

Ok for me to fail all and show error. EKM has no business sending bad keys, it's ok to block the update on it & it will highlight the issue more.

Reply:
> There is a possibility of collision between more than one passphrase frame requests -- one can be coming from a compose frame, and another from the key update. Should we consider emulating such a rare condition in a test and figure out a resolution?

I'd ignore this.

Reply:
Error - toast will do.

Reply:
This is how it looks now (notice the `BrowserMsg` artifact). Is it ok?
![image](https://user-images.githubusercontent.com/23335944/178261326-79dd9b76-cd41-40d2-9751-d9ac013454c7.png)

Reply:
I'm ok with that, rare/unlikely condition. All good